crates/interpreter/src/syntax.rs

```diff
@@ -546,6 +558,50 @@ impl CvtOp {
     }
 }
 
+#[cfg(feature = "threads")]
+macro_rules! impl_atomic_op {
+    ($u:ident) => {
+        paste::paste! {
+            impl AtomicOp {
+                pub fn [< $u >](&self, mem: *mut $u, c: $u) -> $u {
+                    match self {
+                        AtomicOp::Add => unsafe { [< Atomic $u:upper >]::from_ptr(mem).fetch_add(c, Ordering::SeqCst) }
```

Comment:
Those `unsafe { [< Atomic $u:upper >]::from_ptr(mem) }` can be factored out of the match. Such that the match branches only have the relevant operation.

Reply:
Much better, thanks.